radicle-cli/src/commands/auth.rs

```diff
@@ -0,0 +1,226 @@
+        ))
+    }
+}
+
+pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
+    let profile = Profile::load();
+
+    if profile.is_err() {
+        init(options)
+    } else {
+        authenticate(options, ctx)
+    }
+}
+
+pub fn init(options: Options) -> anyhow::Result<()> {
+    term::headline("Initializing your 🌱 profile and identity");
+
+    if git::check_version().is_err() {
+        term::warning(&format!(
+            "Your git version is unsupported, please upgrade to {} or later",
+            git::VERSION_REQUIRED,
+        ));
+        term::blank();
+    }
+
+    let passphrase = term::read_passphrase(options.stdin, true)?;
+    let spinner = term::spinner("Creating your 🌱 Ed25519 keypair...");
+    let profile = Profile::init(passphrase.as_str())?;
+    spinner.finish();
+
+    term::success!(
+        "Profile {} created.",
+        term::format::highlight(&profile.id().to_string())
+    );
+
+    term::blank();
+    term::info!(
+        "Your radicle Node ID is {}. This identifies your device.",
+        term::format::highlight(&profile.id().to_string())
+    );
+
+    term::blank();
+    term::tip!(
+        "To create a radicle project, run {} from a git repository.",
+        term::format::secondary("`rad init`")
+    );
+
+    Ok(())
+}
+
+pub fn authenticate(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
+    let agent = ssh::agent::Agent::connect()?;
+    let profile = match ctx.profile() {
+        Ok(profile) => profile,
+        Err(_) => {
+            anyhow::bail!(
+                "Active identity could not be loaded.\n\
+                To create a new identity, run `rad auth --init`."
+            )
+        }
+    };
+
+    if !options.active {
+        term::info!(
+            "Your active identity is {}",
+            term::format::Identity::new(&profile).styled()
+        );
+    }
+
+    term::headline(&format!(
+        "🌱 Authenticating as {}",
+        term::format::Identity::new(&profile).styled()
+    ));
+
+    let profile = &profile;
+    if !agent.signer(profile.public_key).is_ready()? {
+        term::warning("Adding your radicle key to ssh-agent...");
+
+        // TODO: We should show the spinner on the passphrase prompt,
+        // otherwise it seems like the passphrase is valid even if it isn't.
+        let passphrase = term::read_passphrase(options.stdin, false)?;
+        let spinner = term::spinner("Unlocking...");
+        let mut agent = ssh::agent::Agent::connect()?;
+        let secret = profile
+            .keystore
+            .secret_key(passphrase)?
+            .ok_or_else(|| anyhow!("Key not found in {:?}", profile.keystore.path()))?;
+        agent.register(&secret)?;
+        spinner.finish();
+
+        term::success!("Radicle key added to ssh-agent");
+    } else {
+        term::success!("Signing key already in ssh-agent");
+    };
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use anyhow::Result;
+    use assay::assay;
```

Comment:
Let's get rid of `assay`

Reply:
should I entirely remove tests for now? since it depends on original `common`

Reply:
Yup